packages/aws-cdk-lib/aws-cloudfront/lib/response-headers-policy.ts

```diff
@@ -114,13 +114,36 @@ export class ResponseHeadersPolicy extends Resource implements IResponseHeadersP
       maxLength: 128,
     });
 
+    let securityHeadersBehavior = props.securityHeadersBehavior;
+    let customHeadersBehavior = props.customHeadersBehavior;
+
+    if (securityHeadersBehavior?.contentSecurityPolicy?.reportOnly) {
+      const reportOnlyCSPHeader = {
+        header: 'Content-Security-Policy-Report-Only',
+        value: securityHeadersBehavior.contentSecurityPolicy.contentSecurityPolicy,
+        override: true,
+      };
+      securityHeadersBehavior = {
+        ...securityHeadersBehavior,
+        contentSecurityPolicy: undefined,
+      };
+
+      if (!customHeadersBehavior) {
+        customHeadersBehavior = {
+          customHeaders: [],
+        };
+      }
+      // TODO: log a warning if custom headers already contains CSP-Report-Only header?
```

Comment:
Looks like this PR is not ready?

Reply:
This is mainly a question mark. I didn't notice a pattern for logging at this layer of the application but thought it could be valuable to surface a warning in the event of conflicting configuration. If someone sets `reportOnly`, the expectation would be that there is not already a `Content-Security-Policy-Report-Only` set in the custom headers. However, this cannot be enforced pre-runtime.